src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -138,87 +139,107 @@ def evaluate(
-                                    consumer
-                                ][time_index].values[0]
-            composite_operational_settings[period.start] = composite_operational_setting
-            consumer_system_result.append(
-                self._evaluate_operational_setting(
-                    operational_setting=composite_operational_setting, variables_map=variables_map_for_period
-                )
-            )
+        # Avoid mistakes after this point by not using defaultdict
+        operational_settings_results = dict(operational_settings_results)
+
+        consumer_results = self.collect_consumer_results(operational_settings_used, operational_settings_results)
 
         # Change to 1-index operational_settings_used
         for index, value in enumerate(operational_settings_used.values):
             operational_settings_used[index] = value + 1
 
-        is_valid = reduce(lambda x, y: x.extend(y), [x.component_result.is_valid for x in consumer_system_result])
-        time_steps = reduce(lambda x, y: x.extend(y), [x.component_result.timesteps for x in consumer_system_result])
-        energy_usage = reduce(
-            lambda x, y: x.extend(y), [x.component_result.energy_usage for x in consumer_system_result]
-        )
-        power = reduce(
-            lambda x, y: x.extend(y),
-            [x.component_result.power for x in consumer_system_result if x],
-        )
-
-        consumer_result = ConsumerSystemResult(
+        consumer_system_result = ConsumerSystemResult(
             id=self.id,
             is_valid=is_valid,
-            timesteps=time_steps,
-            energy_usage=energy_usage.fill_nan(0.0),
-            power=power.fill_nan(0.0),
-            operational_settings_results=operational_settings_results,
+            timesteps=sorted({*itertools.chain(*(consumer_result.timesteps for consumer_result in consumer_results))}),
+            energy_usage=reduce(
+                operator.add,
+                [consumer_result.energy_usage for consumer_result in consumer_results],
+            ),
+            power=reduce(
+                operator.add,
+                [consumer_result.power for consumer_result in consumer_results if consumer_result.power is not None],
+            ),
+            operational_settings_results=None,
             operational_settings_used=operational_settings_used,
         )
 
-        # sub_components = list(
-        #     itertools.chain(*[consumer_result.sub_components for consumer_result in consumer_system_result])
-        # )
-
-        models = list(itertools.chain(*[consumer_result.models for consumer_result in consumer_system_result]))
-
         return EcalcModelResult(
-            component_result=consumer_result,
-            sub_components=[],  # Keeping this backward compatible with V1 for now.
-            models=models,
+            component_result=consumer_system_result,
+            sub_components=consumer_results,
+            models=[],
```

Comment:
We should figure out how to solve models for system v2. Currently the core results does not have any parent attribute, so we lose track of which model result belongs to which consumer. 

We also need to merge/collect model results based on operational settings used if we are going to provide model results.

Reply:
you will follow up this in separate PR?

Reply:
I have briefly thought about this problem, so yes, lets figure out